packages/env-utilities/src/lib/env-loader.ts

```diff
@@ -0,0 +1,95 @@
+import { config, type DotenvConfigOptions, type DotenvConfigOutput, type DotenvParseOutput } from 'dotenv';
+import { expand } from 'dotenv-expand';
+import { basename, resolve } from 'node:path';
+
+export interface EnvLoaderOptions extends DotenvConfigOptions {
+	/**
+	 * You may specify a custom environment if `NODE_ENV` isn't sufficient.
+	 */
+	env?: string;
+	/**
+	 * You may specify a required prefix for your dotenv variables (ex. `APP_`).
+	 */
+	prefix?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-inferrable-types
+const packageVersion: string = '[VI]{{inject}}[/VI]';
+
+export function loadEnvFiles(options?: EnvLoaderOptions): DotenvConfigOutput {
+	const log = options?.debug
+		? (message: string) => console.debug(`[@skyra/env-utilities@${packageVersion}][DEBUG] ${message}`)
+		: (_: string) => undefined;
+
+	/**
+	 * @see {@linkplain https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/react-scripts/config/env.js#L18-L23}
+	 */
+	if (!process.env.NODE_ENV) {
+		throw new Error('The NODE_ENV environment variable is required but was not specified.');
+	}
+
+	const env = options?.env || process.env.NODE_ENV;
+	const dotenvPath = options?.path || resolve(process.cwd(), '.env');
+
+	/**
+	 * @see {@linkplain https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/react-scripts/config/env.js#L25-L34}
+	 */
+	const dotenvFiles = [`${dotenvPath}.${env}.local`, `${dotenvPath}.${env}`, dotenvPath];
+
+	if (process.env.NODE_ENV !== 'test') {
+		dotenvFiles.splice(1, 0, `${dotenvPath}.local`);
+	}
+
+	/**
+	 * @see {@linkplain https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/react-scripts/config/env.js#L36-L49}
+	 */
+	let parsed: DotenvParseOutput = {};
+
+	for (const dotenvFile of dotenvFiles) {
+		log(`loading \`${basename(dotenvFile)}\``);
+
+		const result = expand(
+			config({
+				debug: options?.debug,
+				encoding: options?.encoding,
+				path: dotenvFile
+			})
+		);
+
+		if (result.error) {
+			if ((result.error as FSError).code === 'ENOENT') {
+				log(`\`${basename(dotenvFile)}\` file not found`);
+				continue;
+			}
+
+			throw result.error;
```

Comment:
I don't know if it's possible, but you can *maybe* trigger this by creating a file and then denying read permissions using [`fsp.chmod`](https://nodejs.org/dist/latest-v18.x/docs/api/fs.html#fspromiseschmodpath-mode).